### ISSUE TITLE
Unescaped query parameter

### ISSUE THREAD

**Issue by toshipiazza** [input-output-hk/cardano-sl#948]
In Daedalus, [here](https://github.com/input-output-hk/cardano-sl/blob/master/daedalus/src/Daedalus/BackendApi.purs#L45), the query parameters do not seem to be escaped.

Since it looks like we already depend on purescript-contrib, it might be useful to also depend on [purescript-form-urlencoded](https://github.com/purescript-contrib/purescript-form-urlencoded) to fix this problem.

@akegalj

**Comment by akegalj** [comment]
Thanks @toshipiazza . Initial discussion was done over email:

by @toshipiazza :
```
Hi all,

Sorry if I messaged the wrong people, I still have some catching up to
do with names.

As referenced in the chat earlier today, I did a simple once-over of
Daedalus (understanding it's not yet complete) and noticed an
un-escaped query-parameter,
here: https://github.com/input-output-hk/cardano-sl/blob/master/daedalus/src/Daedalus/BackendApi.purs#L45.

This could simply be fixed by escaping the l and r variables. Since I
believe we're already using purescript contrib, it looks like there
exists a purescript-contrib package for url-encoding
strings, https://github.com/purescript-contrib/purescript-form-urlencoded.

I will submit an issue to the issue tracker. Is there a particular
format with which I should submit an issue? All issues I can see are
of the form "[CSL-XXX] <commit message>". Should I use this numbering
scheme as well?

Thanks,
Toshi
```
response by me:
```

> Sorry if I messaged the wrong people, I still have some catching up to
do with names.
I am very sorry if this is stupid question but are you part of one of
IOHK teams ? (its not clear to me at this point)

> As referenced in the chat earlier today
can you point me to the chat. Is it in some IOHK Slack channel ? (if you
are part of the IOHK)

> noticed an un-escaped query-parameter
yes, you are right. We are not escaping query parameters. Most of
sensitive data is sent over HTTP body so there was no use case to escape
it yet. This is a bad practice and we should escape query params.

> This could simply be fixed by escaping the l and r variables. Since I
believe we're already using purescript contrib, it looks like there
exists a purescript-contrib package for url-encoding
strings, https://github.com/purescript-contrib/purescript-form-urlencoded
thank you for the suggestion. I wasn't aware of this package. This is
exactly what fixes the problem

> I will submit an issue to the issue tracker. Is there a particular
format with which I should submit an issue? All issues I can see are of
the form "[CSL-XXX] <commit message>". Should I use this numbering
scheme as well?
You can submit the issue to
https://github.com/input-output-hk/cardano-sl repo . You don't have to
use particular scheme. We use internally numbering scheme because we
have internal issue tracker that manages this. You can just create no
name issue and we will edit the issue accordingly when we mange its
numeration internally (as I am not aware are you part of the IOHK atm)
.  You can assign issue to me (github alieas: akegalj) and we can
continue discussion there.

**Comment by akegalj** [comment]
@toshipiazza this was just merged to cardano-sl-0.5-rc branch https://github.com/input-output-hk/cardano-sl/pull/991#pullrequestreview-45914378 

Thanks for reporting it
